src/v0/destinations/hs/util.js

```diff
@@ -159,6 +158,32 @@ const getTransformedJSON = async (message, destination, propertyMap) => {
           date.setUTCHours(0, 0, 0, 0);
           propValue = date.getTime();
         }
+
+        // Hub spot data type validations 
+        if (propertyMap[hsSupportedKey] === 'string' && typeof propValue !== 'string') {
+          if (typeof propValue === 'object') {
+            propValue = JSON.stringify(propValue);
+          } else {
+            propValue = propValue.toString();
+          }
+        }
+
+        if (propertyMap[hsSupportedKey] === 'bool' && typeof propValue === 'object') {
```

Comment:
what happens if we send random string or number?

Reply:
hubspot is not throwing an error in those cases, it accepts it